src/reason-parser/dune

```diff
@@ -1,42 +1,58 @@
+(rule
+ (targets reason_lexer.mll)
+ (deps reason_lexer.cppo.mll)
+ (action
+  (run %{bin:cppo} -V OCAML:%{ocaml_version} %{deps} -o %{targets})))
+
 (ocamllex
  (modules reason_lexer))
 
 ; Use select.exe from janestreet/ppx_ast to choose lexer_warning.ml based on ocaml version
+
 (rule
  (targets lexer_warning.ml)
- (deps ../generate/select.exe lexer_warning.ml-4.07 lexer_warning.ml-4.06 lexer_warning.ml-default)
- (action (with-stdout-to %{targets} (run ../generate/select.exe lexer_warning.ml-4.07 lexer_warning.ml-4.06 lexer_warning.ml-default))))
+ (deps ../generate/select.exe lexer_warning.ml-4.08 lexer_warning.ml-4.07
```

Comment:
Just for your information, `select.ml` is where we currently make similar compiler version specific decisions. That is an alternative to cppo. (It might not be as easy to do so for these particular special cases in this diff).